File: android/app/src/main/kotlin/com/example/homewidget/PrayerTimeResponse.kt
package com.example.homewidget
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Query

// Data class for API response
data class PrayerTimesResponse(
    val time1: String,
    val time2: String,
    val time3: String,
    val time4: String,
    val time5: String,
    val time6: String,
    val time7: String,
    val date_string: String
)

// Retrofit API Service
interface PrayerTimesApi {
    @GET("solahtimes/api/")
    suspend fun getPrayerTimes(
        @Query("json") json: String = "",
        @Query("lati") latitude: Double,
        @Query("longti") longitude: Double,
        @Query("time-z") timeZone: Int
    ): PrayerTimesResponse
}

// Retrofit Instance
object RetrofitInstance {
    val api: PrayerTimesApi by lazy {
        Retrofit.Builder()
            .baseUrl("https://whiteplus.whitechannel.tv/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()
            .create(PrayerTimesApi::class.java)
    }
}

File: lib/api/res/prayer_time_response.dart
class PrayerTimesResponse {
  final String time1;
  final String time2;
  final String time3;
  final String time4;
  final String time5;
  final String time6;
  final String time7;
  final String dateString;

  // Constructor
  PrayerTimesResponse({
    required this.time1,
    required this.time2,
    required this.time3,
    required this.time4,
    required this.time5,
    required this.time6,
    required this.time7,
    required this.dateString,
  });

  // Factory method: Convert JSON to Object
  factory PrayerTimesResponse.fromJson(Map<String, dynamic> json) {
    return PrayerTimesResponse(
      time1: json["time1"] ?? "",
      time2: json["time2"] ?? "",
      time3: json["time3"] ?? "",
      time4: json["time4"] ?? "",
      time5: json["time5"] ?? "",
      time6: json["time6"] ?? "",
      time7: json["time7"] ?? "",
      dateString: json["date_string"] ?? "",
    );
  }

  // Convert Object back to JSON
  Map<String, dynamic> toJson() {
    return {
      "time1": time1,
      "time2": time2,
      "time3": time3,
      "time4": time4,
      "time5": time5,
      "time6": time6,
      "time7": time7,
      "date_string": dateString,
    };
  }
}